### PR TITLE
Initial service

### PR DESCRIPTION
**Pull request by aleksandar-cakic**
- Updated controllers and queries to accept user_id and friend_id as parameters 
- Tested all routes using postman > all working properly

### CODE REVIEW

File: api/controllers.js [hunk 1]
```diff
@@ -1,111 +1,119 @@
...
       .catch((err) => {
-        res.status(400).send('GET USER PROFILE ERROR: ', err);
+        res.status(400).send(err);
       });
   },
 
   putEditProfile: (req, res) => {
-    return db.putEditProfile(req.query.email, req.body.data.first_name, req.body.data.last_name, req.body.data.age, req.body.data.snack, req.body.data.animal_type)
+    if (!req.query.user_id) {
+      res.status(400).send('Missing parameter "user_id"');
```
Comment (thegraceoflayla) [inline api/controllers.js:27]: Thanks for using such helpful error messages!

File: api/controllers.js [hunk 2]
```diff
@@ -1,111 +1,119 @@
...
-  postFriendFollow: (req, res) => {
-    if (!req.query.userEmail || !req.query.friendEmail) {
-      res.status(400).send('Missing one or more email parameters');
+  friendFollow: (req, res) => {
+    if (!req.query.user_id || !req.query.friend_id) {
+      res.status(400).send('Missing one or more id parameters');
+    }
+
+    if (req.query.user_id === req.query.friend_id) {
+      res.status(200).send('You already are your own best friend! :) \nUnable to follow yourself');
```
Comment (thegraceoflayla) [inline api/controllers.js:87]: 😄

File: service/index.js [hunk 3]
```diff
@@ -25,12 +26,12 @@ const getProfile = (email) => {
...
 };
 
-const putEditProfile = (email, first_name, last_name, age, snack, animal_type) => {
+const putEditProfile = (user_id, first_name, last_name, age, snack, animal_type) => {
   return pool
     .connect()
     .then((client) => {
-      const query = 'UPDATE users SET first_name = $2, last_name = $3, age = $4, snack = $5, animal_type = $6 WHERE email = $1';
-      const values = [email, first_name, last_name, age, snack];
+      const query = 'UPDATE users SET first_name = $2, last_name = $3, age = $4, snack = $5, animal_type = $6 WHERE user_id = $1';
```
Comment (thegraceoflayla) [inline service/index.js:33]: I had not seen this "$" syntax before, but I 'll have to give it a try the next time I'm in Postgres!

**Review by thegraceoflayla — APPROVED** [review]
Everything looks great! Merging now.

File: service/index.js (hunk 3 above)
Comment (aleksandar-cakic) [reply service/index.js:33]: It's really cool, similar to VALUES "?" in mysql

**Pull request merged**